MPTF-65

rkrule opened the pull request:
Per Payments Team Request:

-Add PerformReauthorization to ConfirmFunds Request
-Add ReauthorizationAttempted to ConfirmFunds Reply

Commit:
MPTF-65 - add PerformReauthorization element to Request ConfirmFunds and ReauthorizationAttempted element to Reply ConfirmFunds

ryaan-anthony commented on src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsRequest.php:
```diff
@@ -80,7 +84,7 @@ protected function serializeContents()
             $this->serializePaymentContext() :
             $this->serializePaymentContextBase();
-        return $paymentContext . $this->serializeAmount();
+        return $paymentContext . $this->serializeAmount() . "<PerformReauthorization>{$this->xmlEncode($this->getPerformReauthorization())</PerformReauthorization>";
```
to keep this clean you should return this string value from a protected `serializePerformReauthorization` method

ryaan-anthony commented on src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsReply.php:
```diff
@@ -113,7 +118,7 @@ protected function serializeContents()
             $this->serializePaymentContextBase();
         return $paymentContext .
-        "<FundsAvailable>{$this->xmlEncode($this->getFundsAvailable())}</FundsAvailable>";
+        "<FundsAvailable>{$this->xmlEncode($this->getFundsAvailable())}</FundsAvailable>" . "<ReauthorizationAttempted>{$this->xmlEncode($this->getReauthorizationAttempted())</ReauthorizationAttempted>";
```
to keep this clean you should return this string value from a protected `serializeReauthorizationAttempted` method

ryaan-anthony commented on src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsReply.php:
```diff
@@ -144,4 +149,15 @@ protected function getXmlNamespace()
+
+    public function setReauthorizationAttempted($reauthorizationAttempted)
+    {
+	$this->reauthorizationAttempted = $reauthorizationAttempted;
```
should this be cast as `(bool)`?

ryaan-anthony commented on src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsRequest.php:
```diff
@@ -180,4 +184,15 @@ public function setCurrencyCode($code)
+
+    public function setPerformReauthorization($performReauthorization)
+    {
+	$this->performReauthorization = $performReauthorization;
```
should this be cast as `(bool)`?

ryaan-anthony commented:
You should declare these new methods for the validator: https://github.com/RadialCorp/RetailOrderManagement-SDK/blob/master/src/eBayEnterprise/RetailOrderManagement/Payload/PayloadConfigMap.php#L73

Commit:
fixes per comments from Ryan

Commit:
add optional validator group per Ryan for ConfirmFunds

rkrule commented:
Added protected serialize function for PerformReauth and ReauthAttempted.
Added validator for optional groups for the above properties per ConfirmFunds Request / Reply.
Casted setter property for the above to a boolean.

ryaan-anthony commented:
LGTM
